src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -344,12 +346,13 @@ void Snippet::initOptimalPrimitiveDescriptor() {
         // MatMul has to be decomposed to Brgemm operations before enforcement
         // Note, MatMul decomposition will be run later again for case if BF16 enforcement is not happened
         SNIPPETS_REGISTER_PASS(PassPosition(Place::PipelineStart), ov::snippets::pass::MatMulToBrgemm);
-        SNIPPETS_REGISTER_PASS(PassPosition(Place::After, "MatMulToBrgemm"), pass::EnforcePrecision, element::f32, element::bf16);
+        SNIPPETS_REGISTER_PASS(PassPosition(Place::After, ov::snippets::pass::MatMulToBrgemm::get_type_info_static()),
+                               pass::EnforcePrecision, element::f32, element::bf16);
     }
-
-    SNIPPETS_REGISTER_PASS(PassPosition(Place::Before, "PropagatePrecision"), ov::intel_cpu::pass::BrgemmToBrgemmCPU);
-    SNIPPETS_REGISTER_PASS(PassPosition(Place::Before, "PropagatePrecision"), ov::intel_cpu::pass::SetBrgemmCPUBlockingParams);
-
+    SNIPPETS_REGISTER_PASS(PassPosition(Place::Before, ov::snippets::pass::PropagatePrecision::get_type_info_static()),
```

Comment:
I agree with @a-sidorova: I would prefer to use common instrument - `DiscreteTypeInfo` - for positioned passes matching. Moreover, it automatically prevents potential names collisions (in the case when several transformations from different namespaces have the same names).
Yeah, positioned passes registration looks uglier, but I don't think that this is a big problem because we usually register positioned passes only in one place in the backend.

Reply:
Discussed offline to add a macros to improve readability